awx/main/scheduler/dag_workflow.py

```diff
@@ -215,10 +214,20 @@ def mark_dnr_nodes(self):
             obj = node['node_object']
 
             if obj.do_not_run is False and not obj.job and node not in root_nodes:
-                parent_nodes = [p['node_object'] for p in self.get_dependents(obj)]
+                parent_nodes = [p['node_object'] for p in self.get_parents(obj)]
                 if self._are_all_nodes_dnr_decided(parent_nodes):
                     if self._should_mark_node_dnr(node, parent_nodes):
                         obj.do_not_run = True
                         nodes_marked_do_not_run.append(node)
+            if not obj.do_not_run and obj.all_parents_must_converge:
+                if self._are_relevant_parents_finished(node):
+                    # if the current node is a convergence node and all the 
+                    # parents are finished then check to see if all parents
+                    # met the needed criteria to run the convergence child
+                    # (i.e. parent must fail, parent must succeed)
+                    parent_nodes = [p['node_object'] for p in self.get_parents(obj)]
+                    if (p.do_not_run for p in parent_nodes) or not all(obj in self.get_children(p, p.job.status)for p in parent_nodes):
```

Comment:
I think you need something enveloping `(p.do_not_run for p in parent_nodes)`, like `any`. Otherwise it's an iterator which probably evaluates to True, which is not what you want.

Reply:
I will add that, the desired outcome was achieved without it, but an `any()` would still be optimal